src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets

```diff
@@ -17,6 +17,10 @@
     <_MaxSupportedLangVersion Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(_TargetFrameworkVersionWithoutV)' == '5.0' AND
                                          '$(_MaxSupportedLangVersion)' == ''">9.0</_MaxSupportedLangVersion>
 
+    <!-- .NETCoreApp == 6.0 -->
+    <_MaxSupportedLangVersion Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(_TargetFrameworkVersionWithoutV)' == '6.0' AND
+                                         '$(_MaxSupportedLangVersion)' == ''">10.0</_MaxSupportedLangVersion>
```

Comment:
@chsienki The `LanguageVersionGivenTargetFramework` test was good at catching the new language version :-) 
Can you take a look?

I left `GeneratorDriver` with a `LanguageVersion.Preview` check at the moment (based on offline discussion).